src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidInfiniteRecursion.xml

```diff
@@ -381,6 +381,106 @@ class Issue484_onThisAttributeCall {
     public String foo() {
         return this.other.foo(); // <- false positive
     }
+}
+                     ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>No violations - Issue484-more additional false positives if a parameter is used in the call with calling a method on it, and if any parameter is used in the call in another position</description>
+        <expected-problems>4</expected-problems>
+        <expected-linenumbers>33,43,55,87</expected-linenumbers>
+        <code><![CDATA[
+import java.util.List;
+import my.lib.Request;
```

Comment:
This is an unresolved type? What if all can be resolved: would you be able to compare types in argument lists of methods? (I do no see that in current XPath: check list of types of method params compared to the method definiition list of parameter?).

Reply:
In MethodCall the types are unknown, irrespective of imports. You would have to look them up at definition site.

Reply:
I thought VariableAccess has a Type when defined in class and/or imported?

Reply:
yes, indeed. That you can get it like Node[pmd-java:Type = otherNode/pmd-java:Type] or just as attribute Node/@Type. 
I looked in the pmd own rules xml files, in the  designer source and pmd source but I didn't find a clue that they can be retrieved.
Maybe we should ask in the pmd forum and feature request.